lib/book.rb

```diff
@@ -0,0 +1,29 @@
+class Book
+
+    attr_reader :author_fist_name,
+                :author_last_name,
+                :title,
+                :publication_date
+
+    attr_accessor :checked_out, :rating
+
+    def initialize(data)
+        @author_first_name = data[:author_first_name]
+        @author_last_name = data[:author_last_name]
+        @title = data[:title]
+        @publication_date = data[:publication_date]
+        @checked_out = data[:checked_out] = false
```

Comment:
hmm, but what if this book is at different libraries? Then the @checked_out would need to be specific to each library it's a part of. Probably better for the library to store this data for the books in it's collection.